src/main/play-doc/general/MigrationGuide.md

```diff
@@ -33,18 +33,325 @@ In additional to disabling Elasticsearch via configuration, a service lookup mus
+**From 1.2 onwards, the service ports are deprecated in lieu of [[request ACL|AclConfiguration]].**
 
-```json
-    endpoints        = {
-      "angular-seed-play" = {
-        protocol  = "http"
-        bind-port = 0
-        services  = ["http:/myservice"]
+The [[request ACL|AclConfiguration]] in conjunction with [[dynamic proxy configuration|DynamicProxyConfiguration]] provides additional proxy mapping flexibility on top of what was previously provided by the service ports. This will allow operations to have a complete control on the proxy configuration.
+
+### Migrating HTTP-based services
+
+Some HTTP based service definition can be migrated by supplying its [[request ACL|AclConfiguration]] equivalent, while others require a customized HAProxy configuration.
+
+#### Migrating services definition
+
+The following HTTP based service definition can be migrated by supplying its [[request ACL|AclConfiguration]] equivalent.
+
+##### Services mapped to root path
+
+Given the following example where the service URI was "http://my-service", ConductR will interpret the first path component (`my-service`) as the service name for the purposes of service lookup. By default ConductR will then remove the first path component when rewriting the request. This means that your application or service will be deployed as root context or `/`.
+
+###### Migrating build.sbt
+
+Modify the `Endpoint` declared within the `BundleKeys.endpoints`.
+
+From:
+
+```
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, Set(URI("http://my-service")))
+)
+```
+
+To:
+
+```
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, "my-service",
+    RequestAcl(Http("^/my-service".r -> "/"))
+  )
+)
+```
+
+###### Migrating bundle.conf
+
+Modify the `endpoints` section declared within the `bundle.conf` as such.
+
+From:
+
+```
+    endpoints         = {
+      "endpoint-label" = {
+        bind-protocol = "http"
+        bind-port     = 0
+        services      = ["http://my-service]
+      }
+    }
+```
+
+To:
+
+```
+    endpoints         = {
+      "endpoint-label" = {
+        bind-protocol = "http"
+        bind-port     = 0
+        service-name  = "my-service"
+        acls          = [
+          {
+            http = {
+              requests = [
+                {
+                  path-beg = "/my-service"
+                  rewrite = "/"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    }
+```
+
+
+
+##### Services mapped with preserved path
+
+The `preservePath` query parameter on a service declaration indicated that the context root path should be preserved when passing through ConductR's proxy. Given the service URI `http://my-service?preservePath`, the service is accessible from the outside world on HTTP port allocated by ConductR under `/my-sevice` context.
```

Comment:
@edwardcallahan : paths are not preserved by default - the default behaviour is to have declared service URI rewritten to root context